lib/elixir/pages/Writing Documentation.md

```diff
@@ -70,26 +92,35 @@ When writing documentation:
   * Start new sections with second level Markdown headers `##`. First level headers are reserved for module and function names.
 
   * Place documentation before the first clause of multi-clause functions. Documentation is always per function and arity and not per clause.
+  
+  * Use the `:since` key in the documentation metadata to annotate whenever new functions are added to your API.
 
 ## Doctests
 
 We recommend that developers include examples in their documentation, often under their own `## Examples` heading. To ensure examples do not get out of date, Elixir's test framework (ExUnit) provides a feature called doctests that allows developers to test the examples in their documentation. Doctests work by parsing out code samples starting with `iex>` from the documentation. You can read more about it at `ExUnit.DocTest`.
 
 Notice doctests have limitations. When you cannot doctest a function, because it relies on state or side-effects, we recommend developers include examples directly without the `iex>` prompt.
 
-## Documentation != Comments
+## Documentation != Code Comments
 
 Elixir treats documentation and code comments as different concepts. Documentation is for users of your Application Programming Interface (API), be it your co-worker or your future self. Modules and functions must always be documented if they are part of your API.
 
 Code comments are for developers reading the code. They are useful to mark improvements, leave notes for developers reading the code (for example, you decided not to call a function due to a bug in a library) and so forth.
 
-In other words: documentation is required, code comments are optional.
+Because private functions cannot be accessed externally and effectively exist only in your source code, you can't attach a `@doc` attribute to private functions. However, you can add code comments to private functions, as with any other piece of code, and we recommend developers to do so whenever they believe it will add relevant information to readers and maintainers of the code.
+
+Finally, beware of redundant code comments, such as code comments that are describing the exact same as the code:
+
+    # Total is the sum of the batch and invidual entries
+    total = batch_sum + individual_sum
+
+In summary, documentation are for users of your API (who do not necessarily have access to source) and code comments are for those who interact with the source.
```

Comment:
Not 100% sure about this one, but this sounds better to me

```suggestion
In summary, documentation are for users of your API (who do not necessarily have access to the source code) and code comments are for those who interact with the source code.
```

Reply:
I dont like the use of "might", because I find irrelevant whether you have access to the source or not. If you are reading the documentation, why would you read the source code unless you have a very good reason for it.

Reply:
One more correction:
In summary, documentation **is**